changedetectionio/tests/test_xpath_selector.py

```diff
@@ -277,3 +277,54 @@ def test_check_with_prefix_include_filters(client, live_server):
     assert b"Some text that will change" not in res.data #not in selector
 
     client.get(url_for("form_delete", uuid="all"), follow_redirects=True)
+
+
+def test_non_UTF_8_XPath_double_encoded(client, live_server):
+    res = client.get(url_for("form_delete", uuid="all"), follow_redirects=True)
+    assert b'Deleted' in res.data
+
+    # Give the endpoint time to spin up
+    time.sleep(1)
+
+    # A poorly configured non-utf-8 HTML of server-side.
+    d = b'<html lang="ko">\n<head>\n<met'
+    d += b'a http-equiv="Content-Type" c'
+    d += b'ontent="text/html; charset=EUC'
+    d += b'-KR">\n<style>\np {\n  @charset'
+    d += b' EUC-KR;\n  color: orange;\n  }'
+    d += b'\n</style>\n</head>\n<body>\n<p>'
+    d += b'\xc8\xa5\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd '
+    d += b'\xef\xbf\xbd\xe7\xbf\xac\xef\xbf\xbd\xcf\xb4\xef\xbf\xbd.<'
+    d += b'/p>\n<p>Chaos is natural.</p>\n</body>\n</html>\n'
+
+    with open("test-datastore/endpoint-content.txt", "wb") as f:
+        f.write(d)
+
+    # Add our URL to the import page
+    test_url = url_for('test_endpoint', _external=True)
+    res = client.post(
+        url_for("import_page"),
+        data={"urls": test_url},
+        follow_redirects=True
+    )
+    assert b"1 Imported" in res.data
+    time.sleep(3)
+
+    res = client.post(
+        url_for("edit_page", uuid="first"),
+        data={"include_filters":  "xpath://p", "url": test_url, "tags": "", "headers": "", 'fetch_backend': "html_requests"},
+        follow_redirects=True
+    )
+
+    assert b"Updated watch." in res.data
+    time.sleep(3)
+
+    res = client.get(
+        url_for("preview_page", uuid="first"),
+        follow_redirects=True
+    )
+    # b'\xed\x98\xbc\xe5\x8d\xa0\xec\x8f\x99\xec\x98\x99\xe5\x8d\xa0\xec\x8f\x99\xec\x98\x99 \xe5\x8d\xa0\xec\x8d\xb9\xec\x97\xb0\xe5\x8d\xa0\xec\x8b\xb9\xeb\x8c\x90\xec\x98\x99.' is a wrong encoding result.
+    # Above wrong answer is `answer_below.decode('euc-kr').encode('utf-8')`.
+    assert b'\xc8\xa5\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd \xef\xbf\xbd\xe7\xbf\xac\xef\xbf\xbd\xcf\xb4\xef\xbf\xbd.' in res.data #in selector
```

Comment:
```
# This string is what I wrote.
$ echo '혼돈은 당연하다' | iconv -f UTF-8 -t EUC-KR                       
# This is what iconv shows
ȥ???? ?翬?ϴ?


$ echo '혼돈은 당연하다' | iconv -f UTF-8 -t EUC-KR | iconv -f EUC-KR -t UTF-8
혼돈은 당연하다

# BTW this is but something happens.
# Because as you can see the bytes in the code, somewhere in your program,
# something shreds data.
$ echo 'ȥ���� �翬�ϴ�.' | iconv -f EUC-KR -t UTF-8                             
혼占쏙옙占쏙옙 占썹연占싹댐옙.

-----
# This is what browser shows.
ȥ���� �翬�ϴ�.
# Firefox's repair text encoding feature fixes it correctly.
혼돈은 당연하다.
```

Reply:
https://github.com/dgtlmoon/changedetection.io/assets/61624808/ebf218bd-5987-43f1-bc20-43d1cd756385

Reply:
Also, my example is not as correctly wrong(maybe because it is unknown Korean encoding or something) as I provided the copyrighted html before. However, I proved.

Reply:
I think I found the way.